sdk/identity/identity/src/client/cliCredentialClient.ts

```diff
@@ -0,0 +1,23 @@
+import * as child_process from "child_process";
+
+export interface CredentialClient {
+  createProcess(commandString: string): Promise<any>;
+}
+
+export class CliCredentialClient implements CredentialClient {
+  public createProcess(commandString: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        child_process.exec(commandString, (error, stdout, stderr) => {
+          resolve({ stdout: stdout, stderr: stderr });
+        });
+      } catch (err) {
+        reject(err);
+      }
+    });
+  }
+}
+
+export interface CliCredentialOptions {
+  cliCredentialClient?: CredentialClient;
```

Comment:
Is there a reason you want to make this configurable by the user?

Reply:
Doing so according to the previous [comment](https://github.com/Azure/azure-sdk-for-js/pull/6521#discussion_r362692109)